include/mxnet/tensor_blob.h

```diff
@@ -223,11 +223,11 @@ class TBlob {
    * \param idx the dimension count from the highest dimensin
    * \return the size
    */
-  inline index_t size(index_t idx) const {
+  inline size_t size(index_t idx) const {
```

Comment:
This will break shapes with negative numbers? @reminisce

Reply:
As another effort in parallel, we need to use `-1` to represent unknown dim size to support zero-size tensors. So the type has to be kept as a signed integer.

Reply:
updated